File: 147-insertion-sort-list/insertion-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode insertionSortList(ListNode head) {
        ListNode temp=head;
        while(temp!=null){
            ListNode min=temp;
            ListNode temp2=temp.next;
            while(temp2!=null){
                if(temp2.val<min.val){
                    min=temp2;
                }
                temp2=temp2.next;
            }
            int val=temp.val;
            temp.val=min.val;
            min.val=val;
            temp=temp.next;
        }
        return head;
    }
}

File: 147-insertion-sort-list/insertion-sort-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def insertionSortList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        temp=head
        while temp:
            temp2=temp
            minn=temp
            while temp2:
                if temp2.val<minn.val:
                    minn=temp2
                temp2=temp2.next
            val=minn.val
            minn.val=temp.val
            temp.val=val
            temp=temp.next
        return head